spark/src/main/scala/geotrellis/spark/TileLayerRDDMetadata.scala

```diff
@@ -22,98 +22,14 @@ import geotrellis.tiling._
-  def mapTransform: MapKeyTransform = layout.mapTransform
-  /** TileLayout of the layout */
-  def tileLayout: TileLayout = layout.tileLayout
-  /** Full extent of the layout */
-  def layoutExtent: Extent = layout.extent
-  /** GridBounds of data tiles in the layout */
-  def tileBounds: TileBounds = mapTransform(extent)
-
-  def combine(other: TileLayerMetadata[K])(implicit b: Boundable[K]): TileLayerMetadata[K] = {
-    val combinedExtent       = extent combine other.extent
-    val combinedLayoutExtent = layout.extent combine other.layout.extent
-    val combinedTileLayout   = layout.tileLayout combine other.layout.tileLayout
-    val combinedBounds       = bounds combine other.bounds
-
-    this
-      .copy(
-        extent = combinedExtent,
-        bounds = combinedBounds,
-        layout = this.layout
-          .copy(
-            extent     = combinedLayoutExtent,
-            tileLayout = combinedTileLayout
-          )
-      )
-  }
-
-  def updateBounds(newBounds: Bounds[K])(implicit c: Component[K, SpatialKey]): TileLayerMetadata[K] =
-    newBounds match {
-      case kb: KeyBounds[K] => {
-        val SpatialKey(minCol, minRow) = kb.minKey.getComponent[SpatialKey]
-        val SpatialKey(maxCol, maxRow) = kb.maxKey.getComponent[SpatialKey]
-        val kbExtent = mapTransform(GridBounds(minCol, minRow, maxCol, maxRow))
-
-        kbExtent.intersection(extent) match {
-          case Some(e) =>
-            copy(bounds = newBounds, extent = e)
-          case None =>
-            copy(bounds = newBounds, extent = Extent(extent.xmin, extent.ymin, extent.xmin, extent.ymin))
-        }
-      }
-      case EmptyBounds =>
-        copy(bounds = newBounds, extent = Extent(extent.xmin, extent.ymin, extent.xmin, extent.ymin))
-    }
-}
-
-object TileLayerMetadata {
-  implicit def toLayoutDefinition(md: TileLayerMetadata[_]): LayoutDefinition =
-    md.layout
-
-  implicit def extentComponent[K]: GetComponent[TileLayerMetadata[K], Extent] =
-    GetComponent(_.extent)
-
-  implicit def crsComponent[K]: GetComponent[TileLayerMetadata[K], CRS] =
-    GetComponent(_.crs)
-
-  implicit def layoutComponent[K: SpatialComponent]: Component[TileLayerMetadata[K], LayoutDefinition] =
-    Component(_.layout, (md, l) => md.copy(layout = l))
-
-  implicit def boundsComponent[K: SpatialComponent]: Component[TileLayerMetadata[K], Bounds[K]] =
-    Component(_.bounds, (md, b) => md.updateBounds(b))
-
-  implicit def mergable[K: Boundable]: merge.Mergable[TileLayerMetadata[K]] =
-    new merge.Mergable[TileLayerMetadata[K]] {
-      def merge(t1: TileLayerMetadata[K], t2: TileLayerMetadata[K]): TileLayerMetadata[K] =
-        t1.combine(t2)
-    }
-
-  implicit val tileLayerMetadataFunctor: Functor[TileLayerMetadata] = new Functor[TileLayerMetadata] {
-    def map[A, B](fa: TileLayerMetadata[A])(f: A => B): TileLayerMetadata[B] =
-      fa.copy(bounds = fa.bounds.map(f))
-  }
 
+private[spark] object TileLayerRDDMetadata {
```

Comment:
This doesn't need to be private

Reply:
The file name is suspect after split. Now rather than being container for static methods for `TileLayerMetadata` it's an operation to produce it. `CollectTileLayerMetadata` is the first thing that comes to mind, but that doesn't tag it as belonging to spark. `CollectTileLayerMetadataRDD` is the second but it's certainly not RDD itself.